### PR TITLE
project-schema.json: add ConservationMeasure and conservationMeasure.csv

### LINKED ISSUE #401: conservation measures: create codelist and conservationMeasures object within new environment object
# Background

This issue relates to the following CoST IDS elements proposed in the CoST IDS/OC4IDS review:

<details>

<summary> Conservation measures </summary>

## Conservation measures

**Module:** Environmental and climate
**Indicator:** Biodiversity impact

### Disclosure format

> Disclose and provide further details on the measures adopted by the project to protect and enhance biodiversity. This can comprise, without limitation the following:   
​​avoidance of ecological sitting 
buffers for ecological land 
nature-based solutions 
land restoration 
protection to landscape and historical sites 
invasive species management 
management of wildlife mortality risk 
reduce habitat loss 
pollution reduction 
soil management 
hazardous material management]) 
others (explain)

### OC4IDS mapping

>Project level: For each measure add a `ConservationMeasure` object to the `environment.conservationMeasures` array, setting its `.type` according to the matching code in the open “conservationMeasuresType” codelist and mapping the further explanation to its `.description`. If there is an accompanying document publish in `documents`, with `.documentType` set to 'conservationMeasures’ and include a short description and/or a link to a document providing details. If the details are part of a more general document indicate the relevant section of the document using `.pageStart` and `.pageEnd`. 

</details>

# Proposal

Create a `conservationMeasure` object comprising a `type` and `description`. An array of these measures will be created as `conservationMeasures` within the new `environment` object (see https://github.com/open-contracting/infrastructure/issues/393). `conservationMeasure.type` will take from an open codelist, conservationMeasures.

“conservationMeasures” will be added to the documentType codelist to identify accompanying documentation the publisher may have.

Research is still required to fully define the conservationMeasures codelist.

Add the following fields and objects:

Path | Title | Description | Type (format) | Codelist
-- | -- | -- | -- | --
environment.conservationMeasures | Conservation measures | The measures adopted by the project to protect and enhance biodiversity. | array of `ConservationMeasures` objects | -
ConservationMeasure | Conservation measure | A measure adopted by the project to protect and enhance biodiversity. | object |  
ConservationMeasure.type | Type | The type of conservation measure, using the open conservationMeasures codelist. | string | conservationMeasures
ConservationMeasure.description | Description | A short description of the conservation measure. |   |  

</div>

Add the following codes:

Codelist | Code | Title | Description
-- | -- | -- | --
conservationMeasures | TBD | Avoidance of ecological setting | TBD
conservationMeasures | TBD | Buffers for ecological land | TBD
conservationMeasures | TBD | nature based solutions | TBD
conservationMeasures | TBD | land restoration | TBD
conservationMeasures | TBD | protection to landscape and historical sites | TBD
conservationMeasures | TBD | invasive species management | TBD
conservationMeasures | TBD | management of wildlife mortality risk | TBD
conservationMeasures | TBD | reduce habitat loss | TBD
conservationMeasures | TBD | pollution reduction | TBD
conservationMeasures | TBD | soil management | TBD
conservationMeasures | TBD | hazardous material management | TBD
documentType | conservationMeasures | Conservation measures | Documentation detailing measures to be taken to mitigate the project’s impact on biodiversity

</div>

# Example

```json

{ 
 "environment": { 
   "conservationMeasures": [ 
 	{ 
   	"type": "landRestoration", 
   	"description": "Land restoration measures for the project include..." 
 	} 
   ] 
 }, 
 "documents": [ 
   { 
 	"id": "1", 
 	"documentType": "conservationMeasures", 
 	"url": "http://example.com/conservationMeasures.pdf" 
   } 
 ] 
} 

```

# Sources

* [WWF infrastructure - State of the practice report](https://wwfint.awsassets.panda.org/downloads/wwf_infrastructure_full_report_2018.pdf)
* [WWF KPMG Measuring sustainability in infrastructure investment](https://wwfeu.awsassets.panda.org/downloads/kpmg_full_report.pdf)
* [Testing industry attitudes toward a common reporting approach for ESG data use in infrastructure investment](https://www.guggenheiminvestments.com/GuggenheimInvestments/media/PDF/2022-KPMG-ESG-Data-Report.pdf)
* [Key assessment indicators for teh sustainability of infrastructure projects](https://www.researchgate.net/publication/265623124_Key_Assessment_Indicators_for_the_Sustainability_of_Infrastructure_Projects)
* [Playbook for sustainable infrastructure - WWF](https://issuu.com/fidic/docs/wwf_playbook_-_single_pages?fr=xKAE9_zU1NQ)

cc @evelyndinora

### PR DESCRIPTION
**Pull request by odscjen**
**Related issues**

closes #401 

**Description**

**Merge checklist**

- [x] [Log your changes](https://ocds-standard-development-handbook.readthedocs.io/en/latest/standard/contributing.html#logging-changes)

If there are changes to `project-schema.json` or `project-package-schema.json`:

- [x] Update the examples:
  - [ ] `docs/examples/example.json`
  - [ ] `docs/examples/blank.json`
- [x] Run `./manage.py pre-commit` to update `docs/_static/i8n.csv`

If you added a new definition to the schema, run `./manage.py pre-commit`.

If you added a new codelist:

  - [x] Add an entry to `docs/reference/codelists.md`

### CODE REVIEW

File: schema/project-level/project-schema.json [hunk 1]
```diff
@@ -352,6 +352,17 @@
...
+        "conservationMeasures": {
+          "title": "Conservation measures",
+          "description": "The measures adopted by the project to protect and enhance biodiversity.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConservationMeasure"
+          },
+          "uniqueItems": true,
+          "minItems": 1,
+          "wholeListMerge": true
```
Comment (duncandewhurst) [inline schema/project-level/project-schema.json]: I think we'll need to remove this, depending on the resolution of https://github.com/open-contracting/infrastructure/pull/434#discussion_r1401173099

```suggestion
          "minItems": 1
```

File: docs/reference/codelists.md [hunk 2]
```diff
@@ -83,6 +83,13 @@ Projects with a `status` of 'completed' may be displayed in a list of archived p
 
 ## Open codelists
 
+### conservationMeasure
```
Comment (duncandewhurst) [inline docs/reference/codelists.md]: In keeping with the other headings.

```suggestion
### ConservationMeasure
```

File: schema/project-level/project-schema.json [hunk 3]
```diff
@@ -2370,6 +2381,27 @@
...
+          "description": "The type of conservation measure, using the open [conservationMeasure](https://standard.open-contracting.org/infrastructure/{{version}}/{{lang}}/reference/codelists/#conservationMeasure) codelist.",
+          "type": "string",
+          "codelist": "conservationMeasure.csv",
+          "openCodelist": true,
+          "minLength": 1
+        },
+        "description": {
+          "title": "Description",
+          "description": "A short description of the conservation measure.",
+          "type": "string"
```
Comment (duncandewhurst) [inline schema/project-level/project-schema.json]:
```suggestion
          "type": "string",
          "minLength": 1
```

File: mapping/sustainability.yaml [hunk 4]
```diff
@@ -260,16 +260,16 @@
     mapping: |-
         Project level:
 
-        For each measure add a `ConservationMeasure` object to the `environment.conservationMeasures` array, setting its `.type` according to the matching code in the open "conservationMeasuresType" codelist and mapping the further explanation to its `.description`.
+        For each measure add a `ConservationMeasure` object to the `environment.conservationMeasures` array, set its `.type` according to the matching code in the open [conservationMeasure](../../reference/codelists.md#conservationmeasure) codelist and map the further explanation to its `.description`.
```
Comment (duncandewhurst) [inline mapping/sustainability.yaml]:
```suggestion
        For each measure add a `ConservationMeasure` object to the `environment.conservationMeasures` array, set its `.type` according to the open [conservationMeasure](../../reference/codelists.md#conservationmeasure) codelist and map the further explanation to its `.description`.
```

**Review by duncandewhurst — CHANGES_REQUESTED** [review]
Happy for you to request a review from James once my comments are resolved.

**Comment by odscjen** [comment]
The error is just another example of no `id` field in an array object, but the test is to be changed as per https://github.com/open-contracting/infrastructure/pull/438#discussion_r1403805949

File: schema/project-level/codelists/conservationMeasure.csv [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+natureBasedSolutions,Nature based solutions,"Measures that use solutions already found in nature, for example conserving mangrove forests to protect homes against storm damage."
+habitatRestoration,Habitat restoration,"The restoration of ecosystems negatively impacted by human activities, both historically and as a direct result of the project."
+setAsides,Set-asides,Identification of areas within the project boundaries that are excluded from development and targeted for conservation enhancement measures.
+invasiveSpeciesManagement,Invasive species management,"Measures taken to prevent the introduction of alien, invasive or non-native species, and combat the presence of such species identified within the project boundaries."
+habitatLossReduction,Habitat loss reduction,Measures that reduce the ecological habitats lost as a direct result of the project.
+pollutionReduction,Pollution reduction,"Measures that reduce the level of pollution generated as a direct result of the project. Pollution can include leakage into water courses or soil as well as noise, light, aerosol and gaseous emissions or vibrational nuisance to nearby wildlife populations."
```
Comment (jpmckinney) [inline schema/project-level/codelists/conservationMeasure.csv]:
```suggestion
pollutionReduction,Pollution reduction,"Measures that reduce the pollution generated as a direct result of the project. Pollution can include leakage into water courses or soil as well as noise, light, aerosol and gaseous emissions or vibrational nuisance to nearby wildlife populations."
```

File: schema/project-level/codelists/conservationMeasure.csv [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+soilManagement,Soil management,Measures that integrate sustainable soil management practices according to the Food and Agriculture Organization's Charter and Sustainable Soil Management Criteria.
```
Comment (jpmckinney) [inline schema/project-level/codelists/conservationMeasure.csv]:
```suggestion
soilManagement,Soil management,Measures that integrate sustainable soil management practices. For example, the Food and Agriculture Organization's Charter and Sustainable Soil Management Criteria.
```

File: schema/project-level/codelists/documentType.csv [hunk 7]
```diff
@@ -64,6 +64,7 @@ contractAmendment,Contract amendment,A legal document that changes or adds to th
 designReport,Project design report,"A record of the justification for the design approach adopted in the project, typically issued before the works start, by the consultant or firm that will supervise the works.",OC4IDS
 paymentCertificate,Payment certificate,A document certifying that the work specified on a supplier's estimate or invoice is complete and that payment to the supplier is approved.,OC4IDS
 escalationApproval,Escalation approval,Documentation of one or more approved escalations of the contract price during the project life-cycle.,OC4IDS
+conservationMeasures,Conservation measures,Documentation of measures to be taken to mitigate the project's impact on biodiversity.,OC4IDS
```
Comment (jpmckinney) [inline schema/project-level/codelists/documentType.csv]: The schema is "protect and enhance" rather than "mitigate the project's impact on"

**Review by jpmckinney — APPROVED** [review]
I committed a small change to align the documentType code description with the field description for conservationMeasures

**Pull request merged**